### ISSUE TITLE
UTC time versus local time

### ISSUE THREAD

**Issue by HonzaKr** [echoromeo/pynobo#35]
Hello,
NOBO HUB uses local time, but Home assistant uses UTC time internally and assumes that other integrations also use this time. During the handshake, the UTC time is sent to the NOBO HUB and this will cause a time shift in the switching times in the weekly calendar. It would be possible to fix this.
Thank you

**Comment by oyvindwe** [comment]
Good point about making sure to use the correct time zone in the handshake! However, `pynobo` uses `time.strftime()` to generate the timestamp for the handshake. This does not rely on any Home Assistant configurations, but defaults to the _system_ timezone.

E.g. on my HA box, I have configured the system timezone be Europe/Oslo:
```
s -l /etc/localtime 
lrwxrwxrwx 1 root root 31 Mar 13 06:45 /etc/localtime -> /usr/share/zoneinfo/Europe/Oslo
```

I tried switching timezone in HA and restart it, but system timezone is still being used in the handshake.

I think it would make sense to accept a timezone in the `__init__` method, and, if provided, use this to generate the timestamp in the handshake using e.g. `pytz`. Not that this requires a change both to the `pynobo` library and the HA integration, including adding `pytz` as a dependency in `pynobo`.

Feel free to contribute a PR!

**Comment by oyvindwe** [comment]
It's much simpler than I thought to fix this. HA provides the configured timezone as a `datetime.tzinfo` object.

**Comment by HonzaKr** [comment]
In my opinion, another possibility is to change the handshake in pynobo.py to time.strftime('%Y%m%d%H%M%S', localtime)
br

**Comment by oyvindwe** [comment]
> In my opinion, another possibility is to change the handshake in pynobo.py to time.strftime('%Y%m%d%H%M%S', localtime)

`time.strftime()` uses `time.localtime()` by default, so if your system timezone is set to UTC, that does not help as far as I am able to see.

**Comment by HonzaKr** [comment]
OK.
thx

**Comment by oyvindwe** [comment]
Fix will be included in [HA 2024.4.0](https://github.com/home-assistant/core/milestone/657)
